src/features/app-interfaces/components/create/deployment-details.tsx

```diff
@@ -44,6 +44,7 @@ const baseShape = {
   updatable: z.boolean().optional(),
   deletable: z.boolean().optional(),
 }
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Yeah that'd work fine.

Reply:
I realized that I couldn't do that because I would get the same error. I ended up doing this `type BaseForm = ReturnType<typeof z.object<typeof baseShape>>`